### PR TITLE
Add a workflow task that uses REST

### PR DESCRIPTION
**Pull request by ydayagi**
[FLPATH-144](https://issues.redhat.com//browse/FLPATH-144) https://issues.redhat.com/browse/FLPATH-214
The task can be used as a base class for other/enhanced REST tasks. The protected methods enable that. These methods enable custom building of request headers and custom response processing.

### CODE REVIEW

File: prebuilt-tasks/src/main/java/com/redhat/parodos/tasks/rest/RestWorkFlowTask.java [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+	}
+
+	RestWorkFlowTask(String beanName, RestService restService) {
+		this.restService = restService;
+		setBeanName(beanName);
+	}
+
+	@Override
+	public @NonNull List<WorkParameter> getWorkFlowTaskParameters() {
+		LinkedList<WorkParameter> params = new LinkedList<>();
```
Comment (pkliczewski) [inline prebuilt-tasks/src/main/java/com/redhat/parodos/tasks/rest/RestWorkFlowTask.java:41]: please add TLS support

**Review by pkliczewski — COMMENTED** [review]
The change looks good. Please add tls support

File: prebuilt-tasks/src/main/java/com/redhat/parodos/tasks/rest/RestWorkFlowTask.java [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+	protected HttpEntity<String> buildRequestEntity(WorkContext workContext) throws RestClientException {
+		String content = getOptionalParameterValue(workContext, "content", "");
+		return new HttpEntity<>(content, buildHttpHeaders(workContext));
+	}
+
+	protected HttpHeaders buildHttpHeaders(WorkContext workContext) throws RestClientException {
+		HttpHeaders httpHeaders = new HttpHeaders();
+
+		httpHeaders.setContentType(MediaType.APPLICATION_JSON);
```
Comment (masayag) [inline prebuilt-tasks/src/main/java/com/redhat/parodos/tasks/rest/RestWorkFlowTask.java:109]: should it be configurable?

File: prebuilt-tasks/src/main/java/com/redhat/parodos/tasks/rest/RestWorkFlowTask.java [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+	RestWorkFlowTask(String beanName, RestService restService) {
+		this.restService = restService;
+		setBeanName(beanName);
+	}
+
+	@Override
+	public @NonNull List<WorkParameter> getWorkFlowTaskParameters() {
+		LinkedList<WorkParameter> params = new LinkedList<>();
+		params.add(WorkParameter.builder().key("url").type(WorkParameterType.TEXT).optional(false)
+				.description("URL to send request to").build());
```
Comment (masayag) [inline prebuilt-tasks/src/main/java/com/redhat/parodos/tasks/rest/RestWorkFlowTask.java:43]: not sure how it is in other places, but I'd extract the keys to constants or to a parameter class.
the need to repeat the exact key name and be aware of its type isn't great.

File: prebuilt-tasks/src/main/java/com/redhat/parodos/tasks/rest/RestWorkFlowTask.java (hunk 2 above)
Comment (ydayagi) [reply prebuilt-tasks/src/main/java/com/redhat/parodos/tasks/rest/RestWorkFlowTask.java:109]: we provide basic/general/common config as a base class. user can use it as is or extend.

File: prebuilt-tasks/src/main/java/com/redhat/parodos/tasks/rest/RestWorkFlowTask.java (hunk 3 above)
Comment (ydayagi) [reply prebuilt-tasks/src/main/java/com/redhat/parodos/tasks/rest/RestWorkFlowTask.java:43]: I do not understand what u mean by 'repeat the key' and 'aware of its type'.
The key is defined with its type only once.
The user has to be aware of the type when the request is created.

**Comment by masayag** [comment]
@ydayagi would you like to add integration tests as part of this PR or in a new one and let this be merged?

**Comment by ydayagi** [comment]
> @ydayagi would you like to add integration tests as part of this PR or in a new one and let this be merged?

merge

**Pull request merged**